Consider musl provided built-in defines

Helps compile libtirpc with musl

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

--- ./tirpc/rpc/types.h.orig	2018-03-17 10:23:10.022055255 +0100
+++ ./tirpc/rpc/types.h	2018-03-17 10:23:30.877751656 +0100
@@ -66,7 +66,7 @@
 #define mem_free(ptr, bsize)	free(ptr)
 
 
-#if defined __APPLE_CC__ || defined __FreeBSD__
+#if defined __APPLE_CC__ || defined __FreeBSD__ || !defined(__GLIBC__)
 # define __u_char_defined
 # define __daddr_t_defined
 #endif
